packages/teams-js/test/public/pages.spec.ts

```diff
@@ -156,96 +163,1296 @@ describe('AppSDK-TeamsAPIs', () => {
 
-          await expect(promise).resolves.not.toThrow();
+        utils.respondToMessage(navigateCrossDomainMessage, false);
+
+        await expect(promise).rejects.toThrowError(
+          'Cross-origin navigation is only supported for URLs matching the pattern registered in the manifest.',
+        );
+      });
+    });
+
+    describe('returnFocus', () => {
+      it('should successfully returnFocus', async () => {
+        await utils.initializeWithContext(FrameContexts.content);
+
+        pages.returnFocus(true);
+
+        const returnFocusMessage = utils.findMessageByFunc('returnFocus');
+        expect(returnFocusMessage).not.toBeNull();
+        expect(returnFocusMessage.args.length).toBe(1);
+        expect(returnFocusMessage.args[0]).toBe(true);
+      });
+    });
+
+    describe('registerFocusEnterHandler', () => {
+      it('should successfully register a focus enter handler', async () => {
+        await utils.initializeWithContext('content');
+        pages.registerFocusEnterHandler((x: boolean) => {
+          return true;
         });
-      } else {
-        it(`should not allow calls from ${context} context`, async () => {
-          expect.assertions(1);
-          await utils.initializeWithContext(context);
+        const messageForRegister = utils.findMessageByFunc('registerHandler');
+        expect(messageForRegister).not.toBeNull();
+        expect(messageForRegister.args.length).toBe(1);
+        expect(messageForRegister.args[0]).toBe('focusEnter');
+      });
 
-          await expect(pages.navigateCrossDomain('https://valid.origin.com')).rejects.toThrowError(
-            `This call is only allowed in following contexts: ["content","sidePanel","settings","remove","task","stage","meetingStage"]. Current context: "${context}".`,
-          );
+      it('should successfully invoke focus enter handler', async () => {
+        await utils.initializeWithContext('content');
+
+        let handlerInvoked = false;
+        pages.registerFocusEnterHandler((x: boolean) => {
+          handlerInvoked = true;
+          return true;
         });
+
+        utils.sendMessage('focusEnter');
+        expect(handlerInvoked).toBe(true);
+      });
+    });
+  });
+
+  describe('Frameless - Testing pages module in frameless framework', () => {
+    // Use to send a mock message from the app.
+    const framelessPostMocks = new FramelessPostMocks();
+    const utils = new Utils();
+
+    beforeEach(() => {
+      framelessPostMocks.messages = [];
+      app._initialize(framelessPostMocks.mockWindow);
+    });
+
+    afterEach(() => {
+      // Reset the object since it's a singleton
+      if (app._uninitialize) {
+        app._uninitialize();
       }
     });
 
-    it('should successfully navigate cross-origin', async () => {
-      await utils.initializeWithContext(FrameContexts.content);
+    describe('Testing pages.returnFocus function', () => {
```

Comment:
:nit - I'm not sure it's necessary to preface all the tests with 'testing' since this is a test file, testing is implied

Reply:
I don't think I've seen other PRs use this convention, might be a good thing to bring up on our next team sync to remind folks if this has been decided to be our standard

Reply:
We have discussed this in PR 910 - https://github.com/OfficeDev/microsoft-teams-library-js/pull/910
Can you please have a look at the comments?

Reply:
I can't seem to find the thread on the "Testing *" conversation, would you be able to put up a screenshot of it since deeplinking doesn't work well in Github resolved comment threads? 

Also, I believe you, I think we just have to make sure everyone else follows the convention 😋

Reply:
I agree with @aeunms , I've updated many test files and this is the first that I've heard of this, and I have not followed this convention in any test files, so we should talk about it